File: app/src/main/java/com/apurv/studentassist/accommodation/adapters/UniversitiesListAdapter.java
package com.apurv.studentassist.accommodation.adapters;

import android.content.Context;
import android.graphics.Bitmap;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.android.volley.VolleyError;
import com.android.volley.toolbox.ImageLoader;
import com.apurv.studentassist.R;
import com.apurv.studentassist.accommodation.classes.University;
import com.apurv.studentassist.internet.Network;

import java.util.Collections;
import java.util.List;

/**
 * Created by akamalapuri on 3/15/2017.
 */
public class UniversitiesListAdapter extends RecyclerView.Adapter {


    private LayoutInflater inflater;
    ImageView mImage;

    List<University> mUniversities = Collections.emptyList();

    private ImageLoader mImageLoader;
    private Network network;
    public UniversitiesListAdapter(Context context, List<University> data) {

        inflater = LayoutInflater.from(context);
        this.mUniversities = data;
        network = Network.getNetworkInstnace();
        mImageLoader = network.getmImageLoader();

    }


    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup viewGroup, int viewType) {


        View view = inflater.inflate(R.layout.single_university_view, viewGroup, false);
        RecyclerView.ViewHolder mviewHolder = new UniversitiesViewHolder(view);

        return mviewHolder;

    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder mUniversityRow, int position) {
        if (mUniversityRow instanceof UniversitiesViewHolder) {

            University mUniversity = mUniversities.get(position);


            ((UniversitiesViewHolder) mUniversityRow).universityName.setText(mUniversity.getUniversityName() + " (" + mUniversity.getEstdYear() + ")");
            ((UniversitiesViewHolder) mUniversityRow).UniversityAddress.setText(mUniversity.getLocation());
            ((UniversitiesViewHolder) mUniversityRow).noOfListings.setText(mUniversity.getNoOfListings());
            ((UniversitiesViewHolder) mUniversityRow).noOfUsers.setText(mUniversity.getNoOfUsers());

            loadImages(((UniversitiesViewHolder) mUniversityRow), mUniversity.getUrls().get(0), position);
        }
    }

    private void loadImages(final UniversitiesViewHolder holder, String url, final int position) {


        mImageLoader.get(url, new ImageLoader.ImageListener() {
            @Override
            public void onResponse(ImageLoader.ImageContainer response, boolean isImmediate) {


                Bitmap photo = response.getBitmap();
                if (photo != null) {
                    holder.universityPhoto.setImageBitmap(photo);
                }
            }

            @Override
            public void onErrorResponse(VolleyError error) {

            }
        });

    }

    public void add(University university) {
        mUniversities.add(university);
        notifyItemInserted(mUniversities.size());
    }

    public void pop() {
        mUniversities.remove(mUniversities.size() - 1);
        notifyItemRemoved(mUniversities.size());
    }

    public void addAll(List<University> universityList) {

        for (int i = 0; i < universityList.size(); i++) {
            mUniversities.add(universityList.get(i));
            notifyItemInserted(mUniversities.size());
        }
    }

    public void clear() {

        for (int i = mUniversities.size() - 1; i >= 0; i--) {
            mUniversities.remove(mUniversities.get(i));
            notifyItemRemoved(mUniversities.size());
        }


    }

    @Override
    public int getItemCount() {

        return mUniversities.size();
    }


    class UniversitiesViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        TextView universityName;
        TextView UniversityAddress;
        TextView noOfListings;
        TextView noOfUsers;
        ImageView universityPhoto;


        public UniversitiesViewHolder(View itemView) {
            super(itemView);

            itemView.setOnClickListener(this);
            universityName = (TextView) itemView.findViewById(R.id.universityName);
            UniversityAddress = (TextView) itemView.findViewById(R.id.UniversityAddress);
            noOfListings = (TextView) itemView.findViewById(R.id.noOfListings);
            noOfUsers = (TextView) itemView.findViewById(R.id.noOfUsers);
            universityPhoto = (ImageView) itemView.findViewById(R.id.universityPhoto);
        }

        @Override
        public void onClick(View v) {
            if (v != null) {

            }
        }
    }
}

File: app/src/main/java/com/apurv/studentassist/accommodation/adapters/UniversitiesListAdapter.kt
package com.apurv.studentassist.accommodation.adapters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import com.android.volley.toolbox.ImageLoader
import com.apurv.studentassist.R
import com.apurv.studentassist.accommodation.classes.AccommodationAdd
import com.apurv.studentassist.internet.Network
import kotlinx.android.synthetic.main.single_university_view.view.*

/**
 * Created by akamalapuri on 3/15/2017.
 */
class UniversitiesListAdapter(private val mUniversities: MutableList<AccommodationAdd>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {


    internal var mImage: ImageView? = null

    private val mImageLoader: ImageLoader
    private val network: Network = Network.getNetworkInstnace()


    init {
        mImageLoader = network.getmImageLoader()
    }


    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): RecyclerView.ViewHolder {

        val view = LayoutInflater.from(viewGroup?.context).inflate(R.layout.single_university_view, viewGroup, false)
        return UniversitiesViewHolder(view)

    }

    override fun onBindViewHolder(mUniversityRow: RecyclerView.ViewHolder, position: Int) {
        if (mUniversityRow is UniversitiesViewHolder) {

            (mUniversityRow as UniversitiesViewHolder).onBind(mUniversities.get(position))

//            loadImages(mUniversityRow, mUniversity.getUrls(), position)
        }
    }

//    private fun loadImages(holder: UniversitiesViewHolder, url: String, position: Int) {
//
//
//        mImageLoader.get(url, object : ImageLoader.ImageListener {
//            override fun onResponse(response: ImageLoader.ImageContainer, isImmediate: Boolean) {
//
//
//                val photo = response.bitmap
//                if (photo != null) {
//                    holder.universityPhoto.setImageBitmap(photo)
//                }
//            }
//
//            override fun onErrorResponse(error: VolleyError) {
//
//            }
//        })
//
//    }

//    fun updateList(list: MutableList<University>) {
//        mUniversities = list
//        notifyDataSetChanged()
//    }

    fun add(accommodationAdd: AccommodationAdd) {
        mUniversities.add(accommodationAdd)
        notifyItemInserted(mUniversities.size)
    }

    fun pop() {
        mUniversities.removeAt(mUniversities.size - 1)
        notifyItemRemoved(mUniversities.size)
    }

    fun addAll(universityList: MutableList<AccommodationAdd>) {

        mUniversities.addAll(universityList);
        notifyItemInserted(mUniversities.size);
    }

    fun clear() {

        for (i in mUniversities.indices.reversed()) {
            mUniversities.remove(mUniversities[i])
            notifyItemRemoved(mUniversities.size)
        }


    }

    override fun getItemCount(): Int {

        return mUniversities.size
    }


    internal inner class UniversitiesViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

        fun onBind(add: AccommodationAdd) {


            itemView.noOfRoomsFlashCards.setText(add.noOfRooms)
            itemView.apartmentAddress.setText(add.noOfRooms)
            itemView.cost.setText(add.cost)
            itemView.apartmentNameFlashCards.setText(add.apartmentName)


        }


        override fun onClick(v: View?) {
            if (v != null) {
//                val university = mUniversities[adapterPosition]
//                if (parentActivity.onTouch(university, v)) {
//
//                    university.setSelected(!university.isSelected())
//                    val roundTick = v.findViewById<View>(R.id.selectedPhoto)
//                    Utilities.toggleViewWithRevealAnimation(roundTick)
//                }

            }
        }
    }
}

